src/http/client.cc

```diff
@@ -125,14 +125,15 @@ future<connection::reply_ptr> connection::recv_reply() {
     http_response_parser parser;
     return do_with(std::move(parser), [this] (auto& parser) {
         parser.init();
-        return _read_buf.consume(parser).then([&parser] {
+        return _read_buf.consume(parser).then([this, &parser] {
             if (parser.eof()) {
                 throw std::runtime_error("Invalid response");
             }
 
             auto resp = parser.get_parsed_response();
             sstring length_header = resp->get_header("Content-Length");
             resp->content_length = strtol(length_header.c_str(), nullptr, 10);
+            _keep &= resp->get_header("Connection") != "close";
```

Comment:
Please compare the string in a case-insensitive manner. According to the RFC - see https://www.rfc-editor.org/rfc/rfc7230#section-6.1:

> Connection options are case-insensitive.

Reply:
Fixed